core/src/main/java/org/web3j/protocol/core/DefaultBlockParameter.java

```diff
@@ -10,6 +10,8 @@
  */
 public interface DefaultBlockParameter {
     static DefaultBlockParameter valueOf(BigInteger blockNumber) {
+        if (BigInteger.ZERO.compareTo(blockNumber) >= 0)
+            blockNumber = BigInteger.ZERO;
```

Comment:
Here you should throw an appropriate exception rather than modifying the input.

Reply:
I'm not sure that this did fix the issue - @xaviarias did you have any feedback on it?